### PR TITLE
FAQ Administrator

### PR DESCRIPTION
**Pull request by Anthony01M**
Allows administrators to create, remove or update a FAQ.

### CODE REVIEW

File: server/src/bot/commands/faq-admin.ts [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+                .setRequired(true)
+                .setAutocomplete(true)
+            )
+        )
+    )
+    .listen(async (ctx) => {
+        switch (ctx.interaction.options.getSubcommand()) {
+            case "add": {
+                return ctx.interaction.showModal(addFaqModal(ctx.interaction, [], []))
+            }
```
Comment (0x7d8) [inline server/src/bot/commands/faq-admin.ts:42]: leave an empty line

File: server/src/bot/commands/faq-admin.ts [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+            case "remove": {
+                const faq = ctx.interaction.options.getInteger('faq', true)
+
+                const data = await ctx.database.select({
```
Comment (0x7d8) [inline server/src/bot/commands/faq-admin.ts:47]: dont select unnecessary data, just select id here

File: server/src/bot/commands/faq-admin.ts [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+                })
+
+                await ctx.database.delete(ctx.database.schema.faqs)
+                    .where(eq(ctx.database.schema.faqs.id, faq))
+
+                return ctx.interaction.reply({
+                    ephemeral: true,
+                    content: '`✅` FAQ removed.'
+                })
+            }
```
Comment (0x7d8) [inline server/src/bot/commands/faq-admin.ts:66]: empty line

File: server/src/bot/modals/faq/add-faq.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+                    .setCustomId('content')
+                    .setPlaceholder('content')
+                    .setStyle(TextInputStyle.Paragraph)
+                    .setMinLength(100)
+                    .setRequired(true)
+            ])
+        )
+    )
+    .listen(async (ctx) => {
+        if (!ctx.interaction.guild) return
```
Comment (0x7d8) [inline server/src/bot/modals/faq/add-faq.ts:33]: empty line

File: server/src/bot/modals/faq/add-faq.ts [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+        const title = ctx.interaction.fields.getTextInputValue('title')
+        const content = ctx.interaction.fields.getTextInputValue('content')
```
Comment (0x7d8) [inline server/src/bot/modals/faq/add-faq.ts]: empty line

File: server/src/bot/modals/faq/add-faq.ts [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+                    .setPlaceholder('content')
+                    .setStyle(TextInputStyle.Paragraph)
+                    .setMinLength(100)
+                    .setRequired(true)
+            ])
+        )
+    )
+    .listen(async (ctx) => {
+        if (!ctx.interaction.guild) return
+        const title = ctx.interaction.fields.getTextInputValue('title')
```
Comment (0x7d8) [inline server/src/bot/modals/faq/add-faq.ts]: join these variables with commas, not seperate const declarations

File: server/src/bot/modals/faq/update-faq-content.ts [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+                    .setPlaceholder('content')
+                    .setStyle(TextInputStyle.Paragraph)
+                    .setMinLength(100)
+                    .setRequired(true)
+            ])
+        )
+    )
+    .listen(async (ctx, id: number, title: string) => {
+        if (!ctx.interaction.guild) return
+        const content = ctx.interaction.fields.getTextInputValue('content')
```
Comment (0x7d8) [inline server/src/bot/modals/faq/update-faq-content.ts]: empty line

File: server/src/bot/modals/faq/update-faq-content.ts [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+                    .setCustomId('content')
+                    .setPlaceholder('content')
+                    .setStyle(TextInputStyle.Paragraph)
+                    .setMinLength(100)
+                    .setRequired(true)
+            ])
+        )
+    )
+    .listen(async (ctx, id: number, title: string) => {
+        if (!ctx.interaction.guild) return
```
Comment (0x7d8) [inline server/src/bot/modals/faq/update-faq-content.ts]: empty line

File: server/src/bot/modals/faq/update-faq-content.ts [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+        const content = ctx.interaction.fields.getTextInputValue('content')
+        await ctx.database.update(ctx.database.schema.faqs)
+            .set({ title: title, content: content })
+            .where(eq(ctx.database.schema.faqs.id, id))
```
Comment (0x7d8) [inline server/src/bot/modals/faq/update-faq-content.ts]: empty line

File: server/src/bot/modals/faq/update-faq-title.ts [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+        )
+    )
+    .listen(async (ctx, id: number, content: string) => {
+        if (!ctx.interaction.guild) return
+        const title = ctx.interaction.fields.getTextInputValue('title')
+
+        try {
+            await ctx.database.update(ctx.database.schema.faqs)
+                .set({ title: title, content: content })
+                .where(eq(ctx.database.schema.faqs.id, id))
```
Comment (0x7d8) [inline server/src/bot/modals/faq/update-faq-title.ts]: empty line

File: server/src/bot/modals/faq/update-faq-title.ts [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+                    .setPlaceholder('title')
+                    .setStyle(TextInputStyle.Short)
+                    .setMaxLength(31)
+                    .setMinLength(2)
+                    .setRequired(true)
+            ])
+        )
+    )
+    .listen(async (ctx, id: number, content: string) => {
+        if (!ctx.interaction.guild) return
```
Comment (0x7d8) [inline server/src/bot/modals/faq/update-faq-title.ts]: empty line

**Review by 0x7d8 — CHANGES_REQUESTED** [review]

File: server/src/bot/modals/faq/add-faq.ts [hunk 12]
```diff
@@ -0,0 +1,61 @@
...
+                    .setMinLength(100)
+                    .setRequired(true)
+            ])
+        )
+    )
+    .listen(async (ctx) => {
+        if (!ctx.interaction.guild) return
+
+        const title = ctx.interaction.fields.getTextInputValue('title'),
+        content = ctx.interaction.fields.getTextInputValue('content')
```
Comment (0x7d8) [inline server/src/bot/modals/faq/add-faq.ts]: add a tab before content

**Comment by 0x7d8** [comment]
@Anthony01M for editing, scrap the buttons and just show a modal with title and content but use .setValue to prefill the values with the current state

**Comment by Anthony01M** [comment]
Made it like this so that in case we want to edit just the title or just
the content

**Comment by 0x7d8** [comment]
if you prefill the modal with values that same thing can be achieved better

**Comment by Anthony01M** [comment]
Or we could keep same way and just prefill the details, wouldn't that also
work?

**Comment by 0x7d8** [comment]
please make sure you use tabs, not spaces as identation

**Pull request merged**